// @rafael_et/lumi-sdk/index.js
class Lumi {
  constructor() {
    this.listeners = {};
    
    window.addEventListener("message", (event) => {
      const { type, data, requestId, error } = event.data || {};

      if (!type || !this.listeners[requestId]) return;

      if (error) {
        this.listeners[requestId].reject(error);
      } else {
        this.listeners[requestId].resolve(data);
      }

      delete this.listeners[requestId]; // ✅ cleanup
    });
  }

  send(type, data = {}) {
    const requestId = `req_${Date.now()}_${Math.random()}`;

    return new Promise((resolve, reject) => {
      this.listeners[requestId] = { resolve, reject };

      window.parent.postMessage(
        { type, data, requestId },
        "*"
      );
    });
  }

  // =====================
  // SDK METHODS
  // =====================

  getUser() {
    return this.send("GET_USER");
  }

  pay({ packageId }) {
    return this.send("PAY", { packageId });
  }

  userSpendings({appId}) {
    return this.send("USER_SPENDINGS", { appId });
  }
}

export default Lumi;
